database/migrations/default/1750435376415_ems_person_match_auto/up.sql

```diff
@@ -0,0 +1,461 @@
+CHECK (_match_event_name IS NULL),
+-- add constraint to ensure that crash_pk is never null if person_id is not null
+ADD CONSTRAINT check_person_id_crash_pk_dependency CHECK (
+    (person_id IS NULL) OR (person_id IS NOT NULL AND crash_pk IS NOT NULL)
+);
+
+COMMENT ON COLUMN ems__incidents.person_match_status IS 'The status of the CR3 person record match';
+
+COMMENT ON COLUMN ems__incidents.matched_person_ids IS 'The IDs of the CR3 person records that were found to match this record. Set via trigger when `crash_pk` is not null.';
+
+COMMENT ON COLUMN ems__incidents._match_event_name IS 'Used to dispatch crash matching events. Once dispatched, this value is always set to null via trigger';
+
+--
+-- Function which returns an array of person IDs that match an EMS patient care record
+--
+CREATE
+OR REPLACE FUNCTION public.find_matching_person_ids(
+    ems_record ems__incidents
+) RETURNS integer [] AS $$
+DECLARE
+    matching_person_ids INTEGER[];
+    other_matching_ems_id INTEGER;
+BEGIN
+    raise debug '**function: find_matching_person_ids**';
+    --
+    -- only query for matches when crash_pk is not null, ie the EMS record is already matched to a crash
+    --
+    if ems_record.crash_pk is null then
+        return null;
+    end if;
+    --
+    -- Check to see if there are other EMS patients with the same demographics
+    -- todo: index these columns?
+    --
+    select id from ems__incidents e where
+    e.incident_number = ems_record.incident_number
+    and e.id !=  ems_record.id
+    and e.pcr_patient_age = ems_record.pcr_patient_age
+    and e.pcr_patient_race = ems_record.pcr_patient_race
+    and e.pcr_patient_gender = ems_record.pcr_patient_gender
+    limit 1
+    into other_matching_ems_id;
+    --
+    -- Do not proceed if multiple identical PCRs 
+    --
+    IF other_matching_ems_id IS NOT NULL THEN
+        raise debug 'Multiple EMS patients with same demographic characteristcs found - aborting person match.';
+        return NULL;
+    END IF;
+    --
+    -- Query for matching person records
+    --
+    SELECT
+        COALESCE(array_agg(p.id), ARRAY[]::integer[])
+    FROM
+        people_list_view p
+        left join lookups.drvr_ethncty ethncty_lkp on ethncty_lkp.id = p.prsn_ethnicity_id
+        left join lookups.gndr gndr_lkp on gndr_lkp.id = p.prsn_gndr_id
+    WHERE
+        p.crash_pk = ems_record.crash_pk
+        and p.prsn_age = ems_record.pcr_patient_age
+        AND gndr_lkp.label ilike ems_record.pcr_patient_gender
+        --
+        -- ethnicity can be matched with a case-insensitive wildcard
+        -- except for `American Indian or Alaska Native` which maps
+        -- to the CRIS label `AMER. INDIAN/ALASKAN NATIVE`
+        --
+        AND (
+            ems_record.pcr_patient_race ilike CONCAT('%', ethncty_lkp.label, '%')
+            OR (
+                ethncty_lkp.label ilike '%indian%' 
+                AND ems_record.pcr_patient_race ilike '%indian%'
+            )
+        )
```

Comment:
@roseeichelmann @chiaberry here is the fixed handler for `American Indian or Alaska Native`